### ISSUE TITLE
How to correctly create the connection to a local SQL Server database using TypeORM for Langchain?

### ISSUE THREAD

**Issue by BerkadiaBonal** [typeorm/typeorm#10977]
### What was unclear or otherwise insufficient?

I am trying to connect to my local SQL Server database to use it with Langchain. This is how I am trying to create the connection but keep getting the error shown below when fromDataSourceParams runs:

```
const datasource = new DataSource({

    type: 'mssql',
    host: 'localhost',
    database: 'DB',
    extra: { trustServerCertificate: true },
  });
  const db = await SqlDatabase.fromDataSourceParams({ appDataSource: datasource });
```
```
err = new ConnectionError(err)
ConnectionError: Login failed for user ''.
```

What am I doing wrong when am I creating the connection?

Tried to use my Windows credentials for username and password and be able to connect to the database.

### Recommended Fix

Please give us a clear way of achieving this.

### Are you willing to resolve this issue by submitting a Pull Request?

Yes, I have the time, and I know how to start.

**Comment by walcher** [comment]
Error seems self explanatory. 

ConnectionError: Login failed for user ''.

You should provide a valid user and password for your database.

**Comment by BerkadiaBonal** [comment]
@walcher I am trying to use integrated security which will login with my Windows login. My MSSQL local DB does not have specific credentials. Normally I don't have to provide credentials in connection strings when I configure trustServerCertificate = true with other ORMs. I tried using my Windows credentials but no luck. Here I am not sure why its still asking for credentials.

**Comment by baguskna** [comment]
yes, it looks like you didn't specify the username, based on the docs they have the configuration to connect to db should be like this

`  const datasource = new DataSource({
    type: "mysql",
    host: "0.0.0.0",
    port: "3306",
    username: "root",
    password: "",
    database: "your_db_name",
    synchronize: true,
    logging: false,
  });`
  
  hope this helps
